galasa-ui/src/components/test-runs/TestRunsTabs.tsx

```diff
@@ -90,18 +90,66 @@ export default function TestRunsTabs({runsListPromise, requestorNamesPromise, re
     const columnsOrderParam = columnsOrder.map(col => col.id).join(",");
     
     const params = new URLSearchParams(searchParams.toString());
-    params.set("tab", currentTab.id);
-    params.set("visibleColumns", visibleColumnsParam);
-    params.set("columnsOrder", columnsOrderParam);
+    params.set(PARAMS.TAB, currentTab.id);
+    params.set(PARAMS.VISIBLE_COLUMNS, visibleColumnsParam);
+    params.set(PARAMS.COLUMNS_ORDER, columnsOrderParam);
 
     router.replace(`${pathname}?${params.toString()}`, { scroll: false });
-  }, [selectedVisibleColumns, columnsOrder, isInitialized, pathname, router, searchParams, selectedIndex]);
+  }, [selectedVisibleColumns, columnsOrder, isInitialized, pathname, router, selectedIndex]);
 
   const handleTabChange = (event: {selectedIndex : number}) => {
     const currentIndex = event.selectedIndex;
     setSelectedIndex(currentIndex);
   };
 
+  // Create a canonical query key based on the search parameters so that it won't refetch unnecessarily
+  const queryKey = useMemo(() => {
```

Comment:
Just tried these changes locally and while they do seem to make an improvement, when selecting "All" from the "results" and "status" criteria, no test runs are returned even though there are runs that should be returned.

Reply:
Also noticed an issue when adding tags to the criteria:

If you add a tag that doesn't exist, then save the tags and go to results - no runs are found (this is fine). However, if you go back and add a tag that does exist and should return runs, then no runs are returned.

Similarly, if you add a tag that does exist, then save and go to results - runs will be returned (which is good!). However, if you remove that tag and add a tag that doesn't exist, the runs from the previous results are still shown, which isn't right!

Reply:
Haven't tried it with other search criteria, but this is worth investigating to make sure runs are returned as expected.

Might be worth adding a unit test that performs a couple of interactions, like adding a tag and then removing it, to check that we're sending a request to the API server when we expect it to.